src/index.js

```diff
@@ -379,6 +391,48 @@ class WhatsAppAPI {
         if (!url) throw new Error("URL must be specified");
         return api.authenticatedRequest(this.token, url);
     }
+
+    /**
+     * POST helper, must be called inside the post function of your code.
+     * When setting up the webhook, only subscribe to messages. Other subscritions support might be added later.
+     *
+     * @param {Object} request The request object sent by Whatsapp
+     * @returns {Number} 200, it's the expected http/s response code
+     * @throws {Number} 400 if the POST request isn't valid
+     */
+    post(request) {
+        //Validating payload
+        const signature = request.header("X-Hub-Signature-256");
+        if (!signature) throw 400;
+
+        const hash = createHmac("sha256", this.appSecret)
+            .update(request.rawBody)
+            .digest("hex");
+        if (signature.split("=")[1] !== hash) throw 400;
```

Comment:
I struggled for a week trying to decipher how this check worked. If this works that would make me really happy 😄.

Reply:
I struggled also, it was unclear that besides the token and webhook validation token, the app secret exists.